miniz_zip.c

```diff
@@ -67,18 +67,18 @@ static FILE *mz_freopen(const char *pPath, const char *pMode, FILE *pStream)
 #define MZ_FFLUSH fflush
 #define MZ_FREOPEN mz_freopen
 #define MZ_DELETE_FILE remove
-#elif defined(__MINGW32__)
+#elif defined(__MINGW32__) || defined(__WATCOMC__)
 #ifndef MINIZ_NO_TIME
 #include <sys/utime.h>
 #endif
 #define MZ_FOPEN(f, m) fopen(f, m)
 #define MZ_FCLOSE fclose
 #define MZ_FREAD fread
 #define MZ_FWRITE fwrite
-#define MZ_FTELL64 ftello64
-#define MZ_FSEEK64 fseeko64
-#define MZ_FILE_STAT_STRUCT _stat
-#define MZ_FILE_STAT _stat
+#define MZ_FTELL64 _ftelli64
+#define MZ_FSEEK64 _fseeki64
```

Comment:
With watcom, no problems: it provides its own libc and people almost always statically link to it.

However with mingw: Note that _ftelli64 and _fseeki64 are not available in some old versions of msvcrt.dll.

Reply:
Both MinGW32 and mingw-w64 provide replacement versions of `_ftelli64` and `_fseeki64` for older versions of MSVCRT:
- https://osdn.net/projects/mingw/scm/git/mingw-org-wsl/blobs/5.4-trunk/mingwrt/include/stdio.h
- https://sourceforge.net/p/mingw-w64/mingw-w64/ci/master/tree/mingw-w64-crt/stdio/fseeki64.c

These changes work at least as far back as Windows 98 SE.

Reply:
OK